core/src/term/make/builder.rs

```diff
@@ -0,0 +1,683 @@
+//! A builder interface for producing Nickel records.
+//!
+//! Using this module, you can define Nickel records using a builder style. For example:
+//!
+//! ```rust
+//! # use nickel_lang_core::term::{MergePriority, Term, RichTerm, make::builder::Record};
+//! let b = Record::new()
+//!     .field("foo")
+//!     .priority(MergePriority::Bottom)
+//!     .doc("foo?")
+//!     .not_exported()
+//!     .value(Term::Str("foo".into()));
+//! let t : RichTerm = b
+//!     .field("bar")
+//!     .value(Term::Num(42.into()))
+//!     .into();
+//! ```
+//!
+//! ## Warning
+//!
+//! `RichTerm`s generated by this module are not suitable for evaluation by Nickel. They should
+//! only ever be pretty printed.
```

Comment:
We're ignoring my warning here in the override code in export 😅 I think this was originally just because of fake `TermPos`s that I used. We seem to only use these as part of `Label::default()` now, and it seems we accept that the error reporting might do silly things if it encounters those elsewhere already.
```suggestion
```